packages/office-ui-fabric-react/src/components/SearchBox/SearchBox.tsx

```diff
@@ -145,7 +145,11 @@ export class SearchBox extends BaseComponent<ISearchBoxProps, ISearchBoxState> {
         break;
 
       default:
-        return;
+        if (this.props.onKeyDown) {
```

Comment:
What do you think of this firing first, before the switch. That way you could override the switch if necessary and return before it even happened.

Reply:
I am not sure we want to do that, actually. I guess that a user would expect the `onChange` and `onEscape` callbacks to be invoked at all times when appropriate, and the `onKeyDown` to fire in all other cases. If the callback is invoked before the switch, how would you then define whether the other cases have to checked or not? Would you return a `continueProcessing` flag? That does seem a little bit too complex.

I think that the SearchBox interfaces defines explicitly what callbacks are required to handle which input:
- `onEscape`: Escape pressed
- `onChange`: Text typed
- `onKeyDown`: Any other key pressed